### PR TITLE
feat(plan): add rule to remove filter true nodes

### PR DESCRIPTION
**Pull request by nathanielc**
### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/universe/filter.go [hunk 1]
```diff
@@ -170,3 +173,32 @@ func (t *filterTransformation) UpdateProcessingTime(id execute.DatasetID, pt exe
...
+func (MergeFilterAnyRule) Name() string {
+	return "MergeFilterAnyRule"
+}
+
+func (MergeFilterAnyRule) Pattern() plan.Pattern {
+	return plan.Pat(FilterKind, plan.Any())
+}
+
+func (MergeFilterAnyRule) Rewrite(filterNode plan.PlanNode) (plan.PlanNode, bool, error) {
+	if len(filterNode.Predecessors()) != 1 {
```
Comment (wolffcm) [inline stdlib/universe/filter.go]: You have just one predecessor pattern (`plan.Any()`), so if there were not exactly one predecessor the pattern would not match.  So this check is already enforced by the pattern.

File: stdlib/universe/filter_test.go [hunk 2]
```diff
@@ -580,6 +582,143 @@ func TestFilterOperation_Marshaling(t *testing.T) {
...
+			// from -> filter => from -> filter
+			Rules: []plan.Rule{universe.MergeFilterAnyRule{}},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.PlanNode{
+					plan.CreatePhysicalNode("from", from),
+					plan.CreatePhysicalNode("filter", filterOther),
+				},
+				Edges: [][2]int{{0, 1}},
+			},
+			After: &plantest.PlanSpec{
```
Comment (wolffcm) [inline stdlib/universe/filter_test.go]: These tests can be made a bit more concise by using `RuleTestCase.NoChange` to indicate that no rewrite should be done.

File: stdlib/universe/filter.go (hunk 1 above)
Comment (nathanielc) [reply stdlib/universe/filter.go]: Great I wasn't sure about that so I figured I'd be safe, but I'll update this and the docs.

File: stdlib/universe/filter_test.go (hunk 2 above)
Comment (nathanielc) [reply stdlib/universe/filter_test.go]: great

File: stdlib/universe/filter.go (hunk 1 above)
Comment (nathanielc) [reply stdlib/universe/filter.go]: After a reread of the docs its clear now. I was thinking that more parameters to the Pat function went further up the tree but now I see that it just means more predecessors period. The docs are clear I just had to read them twice.

File: stdlib/universe/filter.go [hunk 3]
```diff
@@ -170,3 +173,29 @@ func (t *filterTransformation) UpdateProcessingTime(id execute.DatasetID, pt exe
 func (t *filterTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
+
+// MergeFilterAnyRule removes Filter nodes whose predicate always evaluates to true.
+type MergeFilterAnyRule struct{}
```
Comment (jlapacik) [inline stdlib/universe/filter.go]: Do you mind renaming this rule to something like `RemoveConstantFilterRule` or `RemoveTrivialFilterRule`? I think something like that would be more descriptive.

**Pull request merged**